src/ucp/core/ucp_request.inl

```diff
@@ -222,6 +222,8 @@ ucp_request_put(ucp_request_t *req)
 static UCS_F_ALWAYS_INLINE void
 ucp_request_complete_send(ucp_request_t *req, ucs_status_t status)
 {
+    ucs_assertv((req->send.ep == NULL) || (req->send.ep->refcount > 0),
```

Comment:
IMO it's weird to check ep refcount during request complete send, it created unneeded dependency

Reply:
but it also helps to find a bug